src/resources/extensions/quarto/docusaurus/docusaurus.lua

```diff
@@ -53,7 +53,8 @@ end
 -- into raw commamark to pass through via dangerouslySetInnerHTML
 local function RawBlock(el)
   if el.format == 'mdx' then
-    return pandoc.CodeBlock(el.text, pandoc.Attr("", {"mdx-code-block"}))
+    -- special mdx-code-block is not handled if whitespace is present after backtrick (#8333)
+    return pandoc.RawBlock("markdown", "````mdx-code-block\n" .. el.text .. "\n````")
```

Comment:
This will break if `el.text` has a line starting with four backticks, right?

Reply:
Yes, but `mdx-code-block` for docusaurus is based on a regex which handles 3 or 4 bacticks.

So 5 backticks for `mdx-code-block` won't work for a .mdx document. 

Based on comment at https://github.com/quarto-dev/quarto-cli/issues/8333#issuecomment-1898410942

https://github.com/facebook/docusaurus/blob/604ccaa89c46dd2e3b515e1beeefa52454447650/packages/docusaurus-utils/src/markdownUtils.ts#L70-L75

````ts
export function unwrapMdxCodeBlocks(content: string): string {
  // We only support 3/4 backticks on purpose, should be good enough
  const regexp3 =
    /(?<begin>^|\n)```mdx-code-block\n(?<children>.*?)\n```(?<end>\n|$)/gs;
  const regexp4 =
    /(?<begin>^|\n)````mdx-code-block\n(?<children>.*?)\n````(?<end>\n|$)/gs;
````

So this is best effort - at the end, this is poor handling on docusaurus side. 

Otherwise, we should not emid `mdx-code-block` by default as it does not seem needed, except on really specific case for some website. And maybe we could find a way to opt-in for a user who needs it.

Reply:
Software was a mistake.